jme3-alloc/src/main/java/com/jme3/alloc/gc/GarbageCollectibleBuffers.java

```diff
@@ -0,0 +1,124 @@
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import com.jme3.alloc.util.NativeBufferUtils;
+
+/**
+ * A collection utility of GC-registered-direct buffers, the collection list is provided through
+ * the registered direct buffers addresses, direct buffers are added to a ReferenceQueue that is 
+ * registered to the GC as a part of post-mortem actions.
+ * 
+ * @author pavl_g
+ */
+public final class GarbageCollectibleBuffers {
+    private final Logger logger = Logger.getLogger(GarbageCollectibleBuffers.class.getName());
+    private final Map<Long, GarbageCollectibleBuffer> bufferAddresses = new HashMap<>();
+    private final ReferenceQueue<Buffer> collectibles = new ReferenceQueue<>();
+    
+    /**
+     * Instantiates a collection of direct buffers that will be registered to be GC'ed.
+     * To start the cleaner thread use {@link GarbageCollectibleBuffers#startMemoryScavenger()}.
+     * 
+     * @see GarbageCollectibleBuffers#register(ByteBuffer)
+     */
+    public GarbageCollectibleBuffers() {
+    }
+    
+    /**
+     * Registers a direct buffer as a {@link GarbageCollectibleBuffer} to the reference queue {@link GarbageCollectibleBuffers#collectibles} 
+     * to be GC'ed as a part of post-mortem actions.
+     * 
+     * @param buffer a buffer to register to the GC reference queue
+     */
+    public void register(Buffer buffer) {
+        GarbageCollectibleBuffer collectibleBuffer = GarbageCollectibleBuffer.from(buffer, collectibles);
+        bufferAddresses.put(collectibleBuffer.getMemoryAddress(), collectibleBuffer);
+    }
+
+    /**
+     * De-allocates a direct buffer using its reference and removes its address 
+     * from the list.
+     * 
+     * @param buffer the buffer memory address
+     * @param isScavenger true if the stack-caller is the {@link MemoryScavenger} thread 
+     *                    associated with this collection, false otherwise.
+     */
+    public void deallocate(Buffer buffer, boolean isScavenger) {
+        if (!buffer.isDirect()) {
+            throw new UnSupportedBufferException("Buffer isn't a direct buffer!");
+        }
+        long bufferAddress = NativeBufferUtils.getMemoryAddress(buffer);
+        deallocate(bufferAddress, isScavenger);
+    }
+
+    /**
+     * De-allocates a direct buffer using its memory address and removes it 
+     * from the list.
+     * 
+     * @param bufferAddress the buffer memory address
+     * @param isScavenger true if the stack-caller is the {@link MemoryScavenger} thread 
+     *                    associated with this collection, false otherwise.
+     */
+    public void deallocate(long bufferAddress, boolean isScavenger) {
+        /* return if the buffer is not in the list of the collectibles */
+        if (!bufferAddresses.containsKey(bufferAddress)) {
+            log(Level.SEVERE, "Buffer " + bufferAddress + " is not found!", isScavenger);
```

Comment:
Future optimization could be done here to prevent unnecessary String objects creation in `"Buffer " + bufferAddress + " is not found!"` if `isScavenger` is true.

Reply:
You can do it 

```
if (!isScavender) {
    logger.log(Level.SEVERE, "Buffer " + bufferAddress + " is not found!");
}
```

no need for  `private void log(Level level, String msg, boolean disabled)` method.

Reply:
I don't prefer nested blocks `{ }`, that is why I haven't suggested it in the first place, but I think it's okay here, the code is not that complex.